### PR TITLE
SC : Refactor Request/Handle event structure

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR refactors service chain request/handle event for next callback feature.
For callback, I will add additional parameter. This PR helps to simplify the code.

- removed event structure
- use original event structure of bridge contract
- rename deposit/withdraw -> request/handle

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by KimKyungup** [comment]
I will clean code tomorrow. :)

File: node/sc/subbridge.go [hunk 1]
```diff
@@ -173,9 +173,10 @@ func NewSubBridge(ctx *node.ServiceContext, config *SCConfig) (*SubBridge, error
 		ctx:            ctx,
 		chainHeadCh:    make(chan blockchain.ChainHeadEvent, chainHeadChanSize),
 		logsCh:         make(chan []*types.Log, chainLogChanSize),
-		//txCh:            make(chan blockchain.NewTxsEvent, transactionChanSize),
-		requestEventCh: make(chan RequestValueTransferEvent, tokenReceivedChanSize),
-		handleEventCh:  make(chan HandleValueTransferEvent, tokenTransferChanSize),
+		//txCh:            make(chan blockchain.NewTxsEvent,
```
Comment (jeonghonah) [inline node/sc/subbridge.go]: How about removing the comment?

File: node/sc/bridge_manager.go [hunk 2]
```diff
@@ -58,31 +58,20 @@ var (
...
-	TokenAddr    common.Address
-	From         common.Address
-	To           common.Address
-	Amount       *big.Int // Amount is UID in ERC721 token
-	RequestNonce uint64
-	URI          string // uri of ERC721 token
-	BlockNumber  uint64
-	Fee          *big.Int
-	txHash       common.Hash
+	*bridgecontract.BridgeRequestValueTransfer
```
Comment (jeonghonah) [inline node/sc/bridge_manager.go]: If `RequestValueTransferEvent` is only pointing the `bridgecontract.BridgeRequestValueTransfer`, how about using the type directly?

File: node/sc/subbridge.go (hunk 1 above)
Comment (KimKyungup) [reply node/sc/subbridge.go]: The comment is not related with this PR. :)
I will consider remove it in another PR.

File: node/sc/bridge_manager.go (hunk 2 above)
Comment (KimKyungup) [reply node/sc/bridge_manager.go]: I tried to use alise type like below.
https://stackoverflow.com/questions/28800672/how-to-add-new-methods-to-an-existing-type-in-go

But, I can't find how to get alias type variable from parent type pointer without hard copy.
I will try to find the way and tell you. :)

**Comment by KimKyungup** [comment]
@jeonghonah @nogang PTAL. :)

File: node/sc/bridge_manager_test.go [hunk 3]
```diff
@@ -179,67 +179,67 @@ func TestBridgeManager(t *testing.T) {
...
 					"to", ev.To.String(),
-					"contract", ev.ContractAddr.String(),
-					"token", ev.TokenAddr.String(),
+					"contract", ev.Raw.Address.String(),
+					"token", ev.ContractAddress.String(),
 					"requestNonce", ev.RequestNonce)
 
-				switch ev.TokenType {
+				switch ev.Kind {
 				case 0:
```
Comment (jeonghonah) [inline node/sc/bridge_manager_test.go]:
```suggestion
				case KLAY:
```

File: node/sc/bridge_manager_test.go [hunk 4]
```diff
@@ -179,67 +179,67 @@ func TestBridgeManager(t *testing.T) {
...
-					tx, err := bridge.HandleKLAYTransfer(&bind.TransactOpts{From: auth2.From, Signer: auth2.Signer, GasLimit: testGasLimit}, ev.Amount, ev.To, ev.RequestNonce, ev.BlockNumber)
+					tx, err := bridge.HandleKLAYTransfer(&bind.TransactOpts{From: auth2.From, Signer: auth2.Signer, GasLimit: testGasLimit}, ev.Amount, ev.To, ev.RequestNonce, ev.Raw.BlockNumber)
 					if err != nil {
-						log.Fatalf("Failed to WithdrawKLAY: %v", err)
+						log.Fatalf("Failed to HandleKLAYTransfer: %v", err)
 					}
 					fmt.Println("WithdrawKLAY Transaction by event ", tx.Hash().Hex())
 					sim.Commit() // block
 
 				case 1:
```
Comment (jeonghonah) [inline node/sc/bridge_manager_test.go]:
```suggestion
				case TOKEN:
```

File: node/sc/bridge_manager_test.go [hunk 5]
```diff
@@ -179,67 +179,67 @@ func TestBridgeManager(t *testing.T) {
...
+					tx, err := bridge.HandleERC20Transfer(&bind.TransactOpts{From: auth2.From, Signer: auth2.Signer, GasLimit: testGasLimit}, ev.Amount, ev.To, tokenAddr, ev.RequestNonce, ev.Raw.BlockNumber)
 					if err != nil {
-						log.Fatalf("Failed to WithdrawToken: %v", err)
+						log.Fatalf("Failed to HandleERC20Transfer: %v", err)
 					}
-					fmt.Println("WithdrawToken Transaction by event ", tx.Hash().Hex())
+					fmt.Println("HandleERC20Transfer Transaction by event ", tx.Hash().Hex())
 					sim.Commit() // block
 
 				case 2:
```
Comment (jeonghonah) [inline node/sc/bridge_manager_test.go]:
```suggestion
				case NFT:
```

File: node/sc/bridge_manager_test.go [hunk 6]
```diff
@@ -179,67 +179,67 @@ func TestBridgeManager(t *testing.T) {
...
-					"token", ev.TokenAddr.String(),
+					"contract", ev.Raw.Address.String(),
+					"token", ev.ContractAddress.String(),
 					"requestNonce", ev.RequestNonce)
 
-				switch ev.TokenType {
-				case 0:
+				switch ev.Kind {
+				case KLAY:
 					// WithdrawKLAY by Event
```
Comment (jeonghonah) [inline node/sc/bridge_manager_test.go]:
```suggestion
```

File: node/sc/bridge_manager_test.go [hunk 7]
```diff
@@ -179,67 +179,67 @@ func TestBridgeManager(t *testing.T) {
...
 					if err != nil {
-						log.Fatalf("Failed to WithdrawKLAY: %v", err)
+						log.Fatalf("Failed to HandleKLAYTransfer: %v", err)
 					}
 					fmt.Println("WithdrawKLAY Transaction by event ", tx.Hash().Hex())
 					sim.Commit() // block
 
-				case 1:
+				case ERC20:
 					// WithdrawToken by Event
```
Comment (jeonghonah) [inline node/sc/bridge_manager_test.go]:
```suggestion
```

File: node/sc/sub_event_handler.go [hunk 8]
```diff
@@ -67,34 +67,34 @@ func (cce *ChildChainEventHandler) HandleLogsEvent(logs []*types.Log) error {
...
+	handleBridgeInfo, ok := cce.subbridge.bridgeManager.GetBridgeInfo(ev.Raw.Address)
 	if !ok {
 		return errors.New("there is no bridge")
 	}
 
 	handleBridgeInfo.UpdateHandledNonce(ev.HandleNonce + 1)
 
-	tokenType := ev.TokenType
-	tokenAddr := ev.TokenAddr
+	tokenType := ev.Kind
```
Comment (jeonghonah) [inline node/sc/sub_event_handler.go]:
```suggestion
	tokenKind := ev.Kind
```

File: node/sc/sub_event_handler.go [hunk 9]
```diff
@@ -67,34 +67,34 @@ func (cce *ChildChainEventHandler) HandleLogsEvent(logs []*types.Log) error {
...
+	tokenAddr := ev.ContractAddress
 
-	logger.Trace("RequestValueTransfer Event", "bridgeAddr", ev.ContractAddr.String(), "handleNonce", ev.HandleNonce, "to", ev.Owner.String(), "valueType", tokenType, "token/NFT contract", tokenAddr, "value", ev.Amount.String())
+	logger.Trace("RequestValueTransfer Event", "bridgeAddr", ev.Raw.Address.String(), "handleNonce", ev.HandleNonce, "to", ev.Owner.String(), "valueType", tokenType, "token/NFT contract", tokenAddr, "value", ev.Value.String())
```
Comment (jeonghonah) [inline node/sc/sub_event_handler.go]:
```suggestion
	logger.Trace("RequestValueTransfer Event", "bridgeAddr", ev.Raw.Address.String(), "handleNonce", ev.HandleNonce, "to", ev.Owner.String(), "valueKind", ev.Kind, "token/NFT contract", ev.ContractAddress, "value", ev.Value.String())
```

File: node/sc/sub_event_handler.go [hunk 10]
```diff
@@ -67,34 +67,34 @@ func (cce *ChildChainEventHandler) HandleLogsEvent(logs []*types.Log) error {
...
 	if !ok {
 		return errors.New("there is no bridge")
 	}
 
 	handleBridgeInfo.UpdateHandledNonce(ev.HandleNonce + 1)
 
-	tokenType := ev.TokenType
-	tokenAddr := ev.TokenAddr
+	tokenType := ev.Kind
+	tokenAddr := ev.ContractAddress
```
Comment (jeonghonah) [inline node/sc/sub_event_handler.go]:
```suggestion
```

File: node/sc/vt_recovery_test.go [hunk 11]
```diff
@@ -570,17 +570,17 @@ func prepare(t *testing.T, vtcallback func(*testInfo)) *testInfo {
 				if ev.RequestNonce == (testTxCount - testPendingCount) {
 					t.Log("missing handle value transfer", "nonce", ev.RequestNonce)
 				} else {
-					switch ev.TokenType {
-					case KLAY, TOKEN, NFT:
+					switch ev.Kind {
+					case KLAY, ERC20, ERC721:
 						break
 					default:
 						t.Fatal("received TokenType is unknown")
```
Comment (jeonghonah) [inline node/sc/vt_recovery_test.go]:
```suggestion
						t.Fatalf("received ev.Kind is unknown: %v", ev.Kind)
```

File: node/sc/bridge_manager.go [hunk 12]
```diff
@@ -520,14 +505,14 @@ func (bm *BridgeManager) LogBridgeStatus() {
 	logger.Info("VT : Service -> Main Chain", "request", s2mTotalRequestNonce, "handle", s2mTotalHandleNonce, "pending", s2mTotalRequestNonce-s2mTotalHandleNonce)
 }
 
-// SubscribeTokenReceived registers a subscription of TokenReceivedEvent.
-func (bm *BridgeManager) SubscribeTokenReceived(ch chan<- RequestValueTransferEvent) event.Subscription {
-	return bm.scope.Track(bm.tokenReceived.Subscribe(ch))
+// SubscribeRequestEvent registers a subscription of TokenReceivedEvent.
```
Comment (jimni1222) [inline node/sc/bridge_manager.go]: How about change 'TokenReceivedEvent' in this comment to requestEventFeeder?

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: reflected :)

File: node/sc/vt_recovery_test.go (hunk 11 above)
Comment (KimKyungup) [reply node/sc/vt_recovery_test.go]: reflected :)

File: node/sc/sub_event_handler.go (hunk 10 above)
Comment (KimKyungup) [reply node/sc/sub_event_handler.go]: reflected :)

File: node/sc/sub_event_handler.go (hunk 9 above)
Comment (KimKyungup) [reply node/sc/sub_event_handler.go]: reflected :)

File: node/sc/sub_event_handler.go (hunk 8 above)
Comment (KimKyungup) [reply node/sc/sub_event_handler.go]: reflected :)

**Pull request merged**